### PR TITLE
Refactor package tooling to support building individual package bundles and multiple repositories

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it

This PR refactors package tooling to support building individual package bundles and repo bundle. This PR also refactors package tooling to support multiple repositories

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/984

### Describe testing done for PR

- to build individual package bundle
`PACKAGE_REPOSITORY=management PACKAGE_NAME=featuregates make package-bundle`
- to build package repo bundle
`PACKAGE_REPOSITORY=management OCI_REGISTRY=${OCI_REGISTRY} make package-repo-bundle`
After running the above two make targets, the build dir would be populated with featuregate package bundle and management packages repo bundle
```
/Users/hyayi/work/tanzu-framework/build
├── package-bundles
│   └── management
│       └── featuregates-v0.10.0-dev.tar.gz
└── package-repo-bundles
    └── management
        ├── packages
        │   └── featuregates.tanzu.vmware.com
        │       ├── 0.10.0-dev.yml
        │       └── metadata.yml
        ├── tanzu-framework-management-repo-0.10.0-dev.tar.gz
        └── tanzu-framework-management-repo-0.10.0-dev.yaml
```
- to build all the package bundles and repo bundle
`OCI_REGISTRY=${OCI_REGISTRY} make package-bundles`
`PACKAGE_REPOSITORY=management OCI_REGISTRY=${OCI_REGISTRY} make package-repo-bundle`
running the above make targets would generate the same build artifacts as above(since we only have one package), except it would build all the packages that are supposed to be in the management package repository
```
/Users/hyayi/work/tanzu-framework/build
├── package-bundles
│   └── management
│       └── featuregates-v0.10.0-dev.tar.gz
└── package-repo-bundles
    └── management
        ├── packages
        │   └── featuregates.tanzu.vmware.com
        │       ├── 0.10.0-dev.yml
        │       └── metadata.yml
        ├── tanzu-framework-management-repo-0.10.0-dev.tar.gz
        └── tanzu-framework-management-repo-0.10.0-dev.yaml
```
### Release note

```release-note
Refactored package tooling to support building individual packages
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by rajathagasthya — APPROVED** [review]

**Review by jmoroski — CHANGES_REQUESTED** [review]
Please complete the "Describe testing done for PR" section of the PR description.

This may just be my ignorance regarding the pipeline, but is there automatic test validation of new make targets or does that need to be included in the change?

**Comment by yharish991** [comment]
@jmoroski 
>Please complete the "Describe testing done for PR" section of the PR description.

Added the testing done for PR

>This may just be my ignorance regarding the pipeline, but is there automatic test validation of new make targets or does that need to be included in the change?

There is no automatic test validation right now, but I think we need to have some testing to validate the scripts(this is something that I will take care of when rewriting these shell scripts in Go)

**Review by anujc25 — APPROVED** [review]
LGTM

**Review by jmoroski — APPROVED** [review]

**Comment by jmoroski** [comment]
@yharish991 

> There is no automatic test validation right now, but I think we need to have some testing to validate the scripts(this is something that I will take care of when rewriting these shell scripts in Go)

please create an issue to track this work.

**Comment by yharish991** [comment]
> please create an issue to track this work.

@jmoroski yes, created an [issue](https://github.com/vmware-tanzu/tanzu-framework/issues/994) to track that work a few days ago

File: packages/package-values.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ managementPackageRepository:
     - name: featuregates
```
Comment (rajathagasthya) [inline packages/package-values.yaml:23]: nit: This file has `.yaml` extension, while kbld config file has `.yml` extension. Can we make it consistent and have it `.yaml`?

**Review by anujc25 — APPROVED** [review]

File: hack/packages/scripts/package-utils.sh [hunk 2]
```diff
@@ -8,86 +8,106 @@ set -eoux pipefail
...
+  REPO_VERSION=${REPO_VERSION:1}
 fi
 
 function generate_single_imgpkg_lock_output() {
-	path=${MANAGEMENT_PACKAGE_REPOSITORY_DIR}/${name}
+	path="${PROJECT_ROOT}/packages/${PACKAGE_REPOSITORY}/${PACKAGE_NAME}"
 	mkdir -p "$path/bundle/.imgpkg"
 	yttCmd="${TOOLS_BIN_DIR}/ytt --ignore-unknown-comments -f $path/bundle/config/"
-	${yttCmd} | "${TOOLS_BIN_DIR}"/kbld -f - -f "${PROJECT_ROOT}/management-packages/kbld-config.yml" --imgpkg-lock-output "$path/bundle/.imgpkg/images.yml" > /dev/null
+	${yttCmd} | "${TOOLS_BIN_DIR}"/kbld -f - -f "${PROJECT_ROOT}/packages/kbld-config.yml" --imgpkg-lock-output "$path/bundle/.imgpkg/images.yml" > /dev/null
```
Comment (anujc25) [inline hack/packages/scripts/package-utils.sh:21]: This needs to change as well: `kbld-config.yml` to `kbld-config.yaml`. And in other places as well.

Comment (yharish991) [reply hack/packages/scripts/package-utils.sh]: thanks for catching that, i forgot to squash and push that commit..

**Pull request merged**